ospf6d/ospf6_interface.c

```diff
@@ -2092,16 +2092,37 @@ DEFUN (ipv6_ospf6_hellointerval,
 	VTY_DECLVAR_CONTEXT(interface, ifp);
 	int idx_number = 3;
 	struct ospf6_interface *oi;
+	uint16_t old_interval;
 	assert(ifp);
 
 	oi = (struct ospf6_interface *)ifp->info;
 	if (oi == NULL)
 		oi = ospf6_interface_create(ifp);
 	assert(oi);
 
+	old_interval = oi->hello_interval;
 	oi->hello_interval = strmatch(argv[0]->text, "no")
 				     ? OSPF_HELLO_INTERVAL_DEFAULT
 				     : strtoul(argv[idx_number]->arg, NULL, 10);
+
+	/*
+	 * If the thread is scheduled and the time remaining would
+	 * take longer than the new interval, then change the hello
+	 * timer to be the new interval, instead of waiting
+	 */
+	if (thread_is_scheduled(oi->thread_send_hello) &&
+	    old_interval != oi->hello_interval) {
+		unsigned long timer_remain =
+			thread_timer_remain_second(oi->thread_send_hello);
+
+		if (timer_remain > oi->hello_interval) {
```

Comment:
Not a blocker, just a question. What if the time passed since the last Hello is already longer than the new timer? Should we send immediately instead of waiting for another one interval?

In the OSPFv2, when the hello timer is changed, a new hello packet is always sent immediately and then a new timer is set. What do you think about doing the same to keep behavior consistent between daemons?

Reply:
If the time left on the timer is greater than the new timer we'll reset the timer then to the new value so there is that.

I didn't send out the hello immediately, but could as well....

Reply:
I'll reset the timer to `now`